queries/yulelads.py

```diff
@@ -139,174 +139,7 @@ def help_text(lemma: str) -> str:
-
-QYuleLadLast/fall →
-    'síðari:lo'_vb_kk_et/fall 'jólasveinn:kk'_gr_et/fall
-
-QYuleLadId/fall →
-    QYuleLadFirst/fall
-    | QYuleLadLast/fall
-    | QYuleLadName/fall
-
-QYuleLadName/fall →
-    {0}
-
-QYuleSuffix →
-    "til" "byggða"
-    | "úr" "fjöllunum"
-    | "til" "bæja"
-    | "til" "borgarinnar"
-
-QYuleDate →
-    "hvenær" QYuleComes/fall QYuleLadId/fall QYuleSuffix?
-
-QYuleDateRel →
-    dagsafs
-
-$score(-4) QYuleDateRel
-
-QYuleNumberOrdinal →
-    raðtala | tala
-
-QYuleValidOrdinal →
-    "ellefta"
-    | "tólfta"
-    | "þrettánda"
-    | "fjórtánda"
-    | "fimmtánda"
-    | "sextánda"
-    | "sautjánda"
-    | "átjánda"
-    | "nítjánda"
-    | "tuttugasta"
-    | "tuttugasta" "og" "fyrsta"
-    | "tuttugasta" "og" "annan"
-    | "tuttugasta" "og" "þriðja"
-    | "tuttugasta" "og" "fjórða"
-
-$score(+4) QYuleValidOrdinal
-
-QYuleTwentyOrdinal →
-    tala
-    | tala "og" QYuleTwentyPart
-
-QYuleTwentyPart →
-    "fyrsta"
-    | "annan"
-    | "þriðja"
-    | "fjórða"
-
-QYuleInvalidOrdinal →
-    "fyrsta"
-    | "annan"
-    | "þriðja"
-    | "fjórða"
-    | "fimmta"
-    | "sjötta"
-    | "sjöunda"
-    | "áttunda"
-    | "níunda"
-    | "tíunda"
-    | "tuttugasta" "og" "fimmta"
-    | "tuttugasta" "og" "sjötta"
-    | "tuttugasta" "og" "sjöunda"
-    | "tuttugasta" "og" "áttunda"
-    | "tuttugasta" "og" "níunda"
-    | "þrítugasta"
-    | "þrítugasta" "og" "fyrsta"
-
-QYuleOrdinal →
-    QYuleNumberOrdinal | QYuleValidOrdinal | QYuleInvalidOrdinal | QYuleTwentyOrdinal
-
-QYuleWhichLad →
-    "hvaða" "jólasveinn"
-    | "hver" "af" "jólasveinunum"
-    | "hver" "jólasveinanna"
-
-QYuleToday →
-    "í" "dag"
-    # | 'í_dag'
-    | "í_kvöld"
-    | "í" "nótt"
-
-QYuleTomorrow →
-    "á_morgun"
-    | "annað" "kvöld"
-    | "aðra" "nótt"
-
-QYuleDay23 →
-    "á" "þorláksmessu"
-
-QYuleDay24 →
-    "á" "aðfangadag"
-
-QYuleDay →
-    "þann"? QYuleOrdinal "desember"?
-    | "þann"? QYuleDateRel
-    | QYuleDay23
-    | QYuleDay24
-    | QYuleToday
-    | QYuleTomorrow
-
-QYuleLadByDate →
-    QYuleWhichLad QYuleComes_nf QYuleSuffix? QYuleDay
-
-QYuleLadFirst →
-    QYuleWhichLad "er" "fyrstur"
-    | QYuleWhichLad "kemur" "fyrstur"
-    | QYuleWhichLad "kemur" "fyrst"
-    | "hver" "er" "fyrsti" "jólasveinninn"
-    | "hver" "er" "fyrstur" "jólasveinanna"
-    | "hvenær" "á" "maður"? "að" "setja" 'skór:kk'_þf "út"? "í" 'gluggi:kk'_et_þf
-    | "hvenær" "setur" "maður" 'skór:kk'_þf "út"? "í" 'gluggi:kk'_et_þf
-    | "hvenær" "fer" 'skór:kk'_et_nf "út"? "í" 'gluggi:kk'_et_þf
-    | "hvenær" "fara" 'skór:kk'_ft_nf "út"? "í" 'gluggi:kk'_et_þf
-
-QYuleLadLast →
-    QYuleWhichLad "er" "síðastur"
-    | QYuleWhichLad "kemur" "síðastur"
-    | QYuleWhichLad "kemur" "síðast"
-    | "hver" "er" "síðasti" "jólasveinninn"
-    | "hver" "er" "síðastur" "jólasveinanna"
-
-QYuleLad →
-    QYuleLadByDate
-    | QYuleLadFirst QYuleSuffix?
-    | QYuleLadLast QYuleSuffix?
-
-""".format(
+GRAMMAR = read_grammar_file("yulelads").format(
```

Comment:
I would suggest that the read_grammar_file handle any potential formatting.

Reply:
Good point, would something like this work? (fargs here meaning format args)
```
def read_grammar_file(filename: str, *fargs, **fkwargs) -> str:
    ...
    return grammar.format(*fargs, **fkwargs)
```
This would allow us to use either index-based formatting like `"{0}".format("something")` or keyword-based formatting `"{replace_me}".format(replace_me="something")`.

Reply:
See changes in commit [caee77d](https://github.com/mideind/Greynir/pull/181/commits/caee77d598b79a13e15cf4af2e2b7f00fec24c68). I added the possibility of specifying format string substitutions as keyword arguments. (and added arithmetic example, showing a bit more complex formatting than yulelads did)

Reply:
As far as I can tell we have three potential strategies when deciding the directory structure accompanying this change.

The first one is your proposed solution, all grammar files in one folder:

queries/module.py
queries/grammar/module.grammar

The second is keeping module and grammar in the same folder:

queries/module.py
queries/module.grammar

The third is keeping each module in its own directory, which contains files required by the module:

queries/module/module.py
queries/module/module.grammar
queries/module/module.js (or whatever)

The second and third solution have the advantage of containing the grammar file next to the module in hierarchical view, but they're both a bit cluttered. The third solution has its merits, but it makes for a more nested hierarchy which can be a PITA. What do you think, @sultur ?

Reply:
I think the second solution would be a bit too cluttered. The third has some benefits, maybe when developing more complex query modules (back-and-forth dialogue maybe), but as you mention the nesting can be a bit tiresome. Therefore I think the first solution is good for the time being.
If we want to later change to the third solution we would also have to modify `query.py` to look for folders correctly, but it wouldn't be too much of a hassle.